docs/scaling-hive.md

```diff
@@ -0,0 +1,47 @@
+# Scaling Hive
+
+This document describes the main places users are likely to see scalability issues with Hive.
+
+## Hive uses CRDs!
+
+Most importantly, be aware that Hive uses CRDs to store its state. The amount of data that Hive can store is limited by the cluster's ability to store CRs. In OpenShift, this boils down to "how much data can we stuff into etcd before the cluster itself becomes unstable?" Best to not chance it. We recommend keeping the total number of Hive CRs in a Hive cluster in the thousands. Anything approaching 10,000 is the danger zone. In practice this means that a single Hive cluster can manage about 1000 clusters.
+
+# Horizontal vs. Vertical Scale
+
+With the exception of install pods (used only when clusters are installing), Hive 1.x is not horizontally scalable at the worker level. Most of the work Hive does happens in the hive-controllers pod, which is one single pod on one single worker. This means that when no installs are running, if you have a cluster with 10 workers, 9 of the workers are very bored. Hive clusters are prime candidates for using worker autoscaling. Keep the worker count as low as you can, but allow bursts of concurrent installs to call for temporary workers to spin up.
+
+In AWS, Hive performs best on c instance types. M instances are fine, but c instances are better.
```

Comment:
Lower case "c" vs upper case "M", is this intentional? I'd sync them up. Also might be good to clarify C is for CPU optimized and M is general purpose.

Reply:
M is capitalized because it's the beginning of a sentence. I'll uppercase the C's too.

Reply:
Or re-organize the sentence to avoid putting M first.
```suggestion
In AWS, Hive performs best on c instance types. Hive performs fine on m instances, but c instances are better.
```

Reply:
I like using the letters, but I did add descriptions in parentheses. Hope this works.